library/threading.c

```diff
@@ -69,69 +69,115 @@ static int threading_mutex_unlock_pthread( mbedtls_threading_mutex_t *mutex )
     return( 0 );
 }
 
-void (*mbedtls_mutex_init)( mbedtls_threading_mutex_t * ) = threading_mutex_init_pthread;
-void (*mbedtls_mutex_free)( mbedtls_threading_mutex_t * ) = threading_mutex_free_pthread;
-int (*mbedtls_mutex_lock)( mbedtls_threading_mutex_t * ) = threading_mutex_lock_pthread;
-int (*mbedtls_mutex_unlock)( mbedtls_threading_mutex_t * ) = threading_mutex_unlock_pthread;
+#define MBEDTLS_MUTEX_INIT_MACRO       threading_mutex_init_pthread
+#define MBEDTLS_MUTEX_FREE_MACRO       threading_mutex_free_pthread
+#define MBEDTLS_MUTEX_LOCK_MACRO       threading_mutex_lock_pthread
+#define MBEDTLS_MUTEX_UNLOCK_MACRO     threading_mutex_unlock_pthread
```

Comment:
This is redundant code and has been removed.

If you'd missed it, I changed the functions to be statically linked when `MBEDTLS_THREADING_PTHREAD` is defined rather than using the function pointers which are now only used when `MBEDTLS_THREADING_ALT` is defined. 

These lines have now been removed.